blackbox/handlers/storage/google_drive.py

```diff
@@ -0,0 +1,252 @@
+                parent_id=parent_id,
+            )
+        else:
+            return folder["id"]
+
+    def _upload(self, file_path: str, file_content: bytes | str) -> str:
+        """
+        Upload a file to Google Drive.
+
+        Args
+            file_path: The path to upload the file to.
+            file_content: The content of the file to upload.
+    
+        Return
+            The ID of the uploaded file.
+        """
+
+        # Determine the MIME type of the file, because we need to include this in the
+        # payload when we upload the file to Google Drive.
+        mimetype, _ = mimetypes.guess_type(file_path)
+
+        # Get the folder
+        folder_id = "root"  # Use the root folder as a default
+        folder_path = "/".join(file_path.split("/")[:-1])  # Path excluding filename
+        if self.upload_base:
+            folder_id = self._get_and_ensure_deepest_folder_id(path=folder_path)
+
+        # Prepare the file metadata
+        file_name = file_path.split("/")[-1]
+        file_metadata = {"name": file_name, "parents": [folder_id]}
+
+        # Upload the file to Google Drive
+        # The size of one chunk. Larger files will be uploaded in multiple chunks.
+        chunk_size = 4950000  # 5 MB limit, let's stay a bit under for safety
+        file_io = BytesIO(file_content)
+        media = MediaIoBaseUpload(
+            file_io,
+            chunksize=chunk_size,
+            mimetype=mimetype,
+            resumable=True,  # Allow this upload to occur in multiple parts
+        )
+        response = self.client.files().create(
+            body=file_metadata,
+            media_body=media,
+            fields="id",
+        ).execute()
+        return response["id"]
+
+    def sync(self, file_path: Path) -> None:
+        """Sync a file to Google Drive."""
+        # Compress the file and build the destination file path
+        temp_file, recompressed = self.compress(file_path)
+        ext = ".gz" if recompressed else ""
+        upload_path = f"{self.upload_base}/{file_path.name}{ext}"
+        try:
+            with temp_file as f:
+                # Upload the file
+                self._upload(file_path=upload_path, file_content=f.read())
+            self.success = True
+        except HttpError as e:
+            log.error(e)
+            self.success = False
+            self.output = str(e)
+
+    def rotate(self, database_id: str) -> None:
+        """
+        Rotate the files in the Google Drive directory.
+
+        All files in the base directory of backups will be deleted when they are older
+        than `retention_days`. Because of this, it's better to store backups in an
+        isolated folder.
+        """
+
+        retention_days = 7  # Default
+        if Blackbox.retention_days:
+            retention_days = Blackbox.retention_days  # Overwrite if this is configuired
+
+        # Get the folder (remove the trailing slash from the base folder)
```

Comment:
confused by this comment, doesn't seem like we _do_ remove the trailing slash. Or is this a request for the user and not an imperative comment?